File: src/models/inscricao_treinamento.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Date, DateTime
from src.models import db


class InscricaoTreinamento(db.Model):
    __tablename__ = 'inscricoes_treinamento_planejamento'
    id = Column(Integer, primary_key=True)
    treinamento_id = Column(Integer, nullable=True)
    nome_treinamento = Column(String(255), nullable=False)
    matricula = Column(String(50), nullable=False)
    tipo_treinamento = Column(String(100), nullable=False)
    nome_completo = Column(String(255), nullable=False)
    naturalidade = Column(String(120), nullable=False)
    email = Column(String(255), nullable=False)
    data_nascimento = Column(Date, nullable=False)
    cpf = Column(String(14), nullable=False)
    empresa = Column(String(255), nullable=False)
    criado_em = Column(DateTime, default=datetime.utcnow)

File: src/static/js/inscricao_treinamento.js
(function () {
  const params = new URLSearchParams(window.location.search);
  const treinamentoId = params.get('treinamentoId');
  const nomeQS = params.get('nome');

  const form = document.getElementById('formInscricao');
  const submitButton = document.getElementById('btnEnviar');
  const nomeTreinamentoInput = document.getElementById('nomeTreinamento');

  if (nomeQS) {
    nomeTreinamentoInput.value = decodeURIComponent(nomeQS);
  }

  form.addEventListener('submit', async (event) => {
    event.preventDefault();

    const validator = window.FormValidation?.get(form);
    if (validator && !validator.validate()) {
      return;
    }

    const dataNascimentoIso = window.FormValidation
      ? window.FormValidation.sanitizeField(document.getElementById('dataNascimento'))
      : document.getElementById('dataNascimento').value;

    const payload = {
      matricula: document.getElementById('matricula').value?.trim(),
      treinamento_id: treinamentoId,
      nome_treinamento: nomeTreinamentoInput.value,
      tipo_treinamento: document.getElementById('tipoTreinamento').value?.trim(),
      nome_completo: document.getElementById('nomeCompleto').value?.trim(),
      naturalidade: document.getElementById('naturalidade').value?.trim(),
      email: document.getElementById('email').value?.trim(),
      data_nascimento: dataNascimentoIso,
      cpf: window.FormValidation
        ? window.FormValidation.sanitizeField(document.getElementById('cpf'))
        : document.getElementById('cpf').value?.trim(),
      empresa: document.getElementById('empresa').value?.trim(),
    };

    await executarAcaoComFeedback(submitButton, async () => {
      await chamarAPI('/api/inscricoes-treinamento', 'POST', payload);
      alert('Inscrição enviada com sucesso!');
      window.location.href = '/planejamento-treinamentos.html';
    });
  });
})();
